File: Algorithms and Data Structures/Practice/0 - Other/product-eq.js
'use strict';

function product(x, i) {
    let p = 1;
    for (let k = 1; k <= i; k++) {
        p = p * Math.cos(x / (2 ** k));
    }
    return p;
}

const x = 1.5;
const res = product(x, 5);
console.log(res);

File: Algorithms and Data Structures/Practice/0 - Other/product-eq.c
#include <stdio.h>
#include <math.h>

#define N 5
int main() {

    double X[N] = {0};
    double x = 1.6;

    for (int i = 1; i < N; i++) {
        double P = 1.0; 
        for (int k = 1; k <= i; k++) {
            P *= cos(x/ pow(2, k));
        }
        X[i] = P;
    }

    for (int i = 0; i < N; i++) {
        printf("%f ", X[i]);
    }

    return 0;
}